### ISSUE TITLE
Infinite Loop Encountered in JMARS in mouseMoved Event

### ISSUE THREAD

**Issue by zanderson9** [vincenzopalazzo/material-ui-swing#122]
This issue may have been fixed in release 1.1.1_0.1, but I want to describe it anyway:

JMARS is currently using version 1.1.1 **before** pre-release 6.0

In certain scenarios, users encounter OutOfMemoryError, where stack trace looks like in the attached log file.

As seen in that log, there is an infinite loop of this event:

**_java.awt.AWTEventMulticaster.mouseMoved(AWTEventMulticaster.java:329)_**

It also appears that this library debug output occurs before AWTMultiCaster occurs:
**"component button set color"**

My investigation shows that this debug output is in MaterialUITimer.java class, line #163, in the version of the library that JMARS is currently using, 1.1.1 before pre-release 6.0.

However, in the source code for MaterialUITimer.java which is on the "**master**" branch in GIT, line #163 and the following lines up to line #168 are commented out.

Also, trying 1.1.1_0.1, even for a short time, seems to not have this infinite loop effect.

Question: are the lines 163-168 removed in version 1.1.1_0.1?

[jmars_beta_error_log_crop2.txt](https://github.com/vincenzopalazzo/material-ui-swing/files/4362061/jmars_beta_error_log_crop2.txt)
 the attached file

**Comment by vincenzopalazzo** [comment]
Hi @zanderson9,

I should know if this problem was generated from a JButton with a default implementation or a personal implementation.

If the exception was generated to personal implementation, I think the error is inside the personal implementation, is very difficult to work with a personal button with the material mouse hover effect.

if the exception was generated to a button with default implementation, I think the bug is the implementation inside the MaterialButtonUI.

I worked hard on this problem of integration between MaterialButtonUI and MaterialUITimer and after a few months of work I decided to refactorize the MaterialButtonUI with a new implementation and from the version > pre-release6.0 the button have an implementation without know bug.

The new implementation of the JButton style is available from the version >= pre-release6.0 but only in the version > pre-release6.1 the implementation have all know bug fixed.

So, I don't know if the bug now was fixed, maybe yes, maybe no, but we can test it inside the jMars and have the feedback from my work.

I hope this answer can help you and your work
